src/ast/passes/semantic_analyser.cpp

```diff
@@ -1767,7 +1788,11 @@ void SemanticAnalyser::visit(Map &map)
     // If there is no record of any assignment after the first pass
     // then it's safe to say this map is undefined
     if (!is_first_pass()) {
-      map.addError() << "Undefined map: " << map.ident;
+      if (initializer) {
+        map.addError() << "Unused map: " << map.ident;
```

Comment:
I think this should be `addWarning`. I don't object to it being error for now, but in the future, such a map declaration could be for the purposes of an external object, etc.

Reply:
> such a map declaration could be for the purposes of an external object

Yeah i forgot about that. I'm fine with making it a warning.